### DISCUSSION TITLE
hook before/after sending the email

### DISCUSSION THREAD

**Discussion by jorythompson** [berrystephenw/OctoText#89] [General]
I finally got this plugin to work (probably user-error, but I won't admit it ;-)
Is there any way to have a hook before taking the snapshot so I can turn on the light, and one after so I can turn it off?  I have a light controlled by my raspberry pi (via octolight) and can also drive the light via a shell script and gpio.

thanks in advance

**Comment by berrystephenw** [comment]
Hi Jory. I'm sure that it is possible to do that - I'm just not familiar with how to - and I don't have a way to test it. It would probably be easier to modify octolight to react to printer events than to modify octotext to run a script. 
There are plugins that will do what you want - I just checked out  [WLED](https://plugins.octoprint.org/plugins/wled/) and it looks like it would do exactly what you are asking. 

Steve

**Reply by jorythompson** [reply]
Looks like WLED will react to specific events.  What event does OctoText initiate that WLED can look for?  

I would think that it wouldn't be difficult for OctoText to call a BASH script before and after initiating an email.  Let the user write the scripts to handle whatever they want them to do.

**Reply by berrystephenw** [reply]
Octotext reacts to the same events, octoprint sends the printer events to all the plugins (start, done, error etc) depending on which ones you have enabled. In your case you probably want to turn on the lights with either file upload or start and off with done. The plug-in author is very active on the forums and discord I'm sure he could help you if you have trouble. His user name is cp2004 and he helped me with octotext development.
